### ISSUE TITLE
Tests should show error more explicitly

### ISSUE THREAD

**Issue by bquistorff** [gvegayon/parallel#32]
Right now, the user has to manual check `all_tests_results.txt` for errors. (I think by searching for `with error` and `r(`). It would be nice if this was somehow automatic. We could:

- have `make` could automatically search through the output for errors
- have the stata run return appropriate error codes. We would explicitly error out of Stata on parallel errors by adding _assert(...), msg(...) after the calls to `parallel` in the test files. Stata doesn't set its return code properly on error, so this wouldn't get automatically propagated back to make, but we could have a stata executable wrapper that does this on it own like [this](https://gist.github.com/pschumm/b967dfc7f723507ac4be) (I've made and used updated versions of this).

**Comment by gvegayon** [comment]
I don't think this is crucial (we shouldn't get any errors from the tests!... right?). But if you can make it work go ahead =).
